content/en/blog/2024/elastic-contributes-continuous-profiling-agent.md

```diff
@@ -0,0 +1,148 @@
+  Wert](https://github.com/AlexanderWert) (Elastic), [Morgan
+  McLean](https://github.com/mtwo) (Splunk), [Ryan
+  Perry](https://github.com/Rperry2174) (Grafana)
+draft: true # TODO: remove this line once your post is ready to be published
+# canonical_url: http://somewhere.else/ # TODO: if this blog post has been posted somewhere else already, uncomment & provide the canonical URL here.
+issue: https://github.com/open-telemetry/community/issues/1918
+sig: Profiling SIG
+---
+
+Following significant collaboration between
+[Elastic](https://www.elastic.co/observability-labs/blog/elastic-donation-proposal-to-contribute-profiling-agent-to-opentelemetry)
+and [OpenTelemetry's profiling community](./profiling.md), which included a
+thorough review process, we’re excited to announce that the OpenTelemetry
+project has accepted
+[Elastic's donation of its continuous profiling agent](https://github.com/open-telemetry/community/issues/1918).
+
+This marks a significant milestone in establishing profiling as a core telemetry
+signal in OpenTelemetry. Elastic’s [eBPF](https://ebpf.io/) profiling agent
+observes code across different programming languages and runtimes, third-party
+libraries, kernel operations, and system resources with low CPU and memory
+overhead in production. Both, SREs and developers can now benefit from these
+capabilities: quickly identifying performance bottlenecks, maximizing resource
+utilization, reducing carbon footprint, and optimizing cloud spend.
+
+Elastic’s decision to contribute the project to OpenTelemetry was made to
+accelerate OpenTelemetry’s mission and enable effective observability through
+high-quality, portable telemetry. This collaboration also shows the commitment
+to vendor neutrality and community-driven development enhancing the overall
+profiling and observability ecosystems.
+
+The Elastic team would like to thank the OpenTelemetry community for the great
+and constructive cooperation on the donation proposal. We look forward to
+jointly establishing continuous profiling as an integral part of OpenTelemetry.
+
+With today’s acceptance, Elastic’s continuous profiling agent will be
+contributed to OpenTelemetry. This agent will now be jointly supported by both
+Elastic’s team as well as a diverse set of official maintainers from different
+companies:
+
+- Dmitry Filimonov (Grafana Labs)
+- Felix Geisendörfer (Datadog)
+- Jonathan Halliday (Red Hat)
+- Christos Kalkanis (Elastic)
+
+## What is continuous profiling?
+
+[Continuous profiling](https://www.cncf.io/blog/2022/05/31/what-is-continuous-profiling/)
+is a technique used to understand the behavior of a software application by
+collecting information about its execution. This includes tracking the duration
+of function calls, memory usage, CPU usage, and other system resources along
+with associated metadata.
+
+## Benefits of Continuous Profiling
+
+Traditional profiling solutions have significant drawbacks limiting adoption in
```

Comment:
What is "Traditional profiling", a reader new to this topic _might_ not know the difference between the one and the other. Is there an easy way to explain in a few words what this is and by that make it clearer where the drawbacks are coming from?

Reply:
I'd say first define "Continuous Profiling" properly in the previous section.
Then "Traditional Profiling" is any other solution (non-continuous profiling).
So maybe just replace `Traditional` with `Other`?

Reply:
And then I think the explanation of the differences is clarified in the following text.

Reply:
I think "Traditional" is better than "Other", or even "Non-continuous profiling" would be fine, but ultimately we need to make clear what we mean by that, because from that the drawbacks become clear, otherwise we will have a hard time to argue that every "other" profiling is really not able to attach to a service at runtime, or is every "other" profiling solution really not able to get visibility into third-party libraries?

Reply:
> Traditional profiling solutions, typically used for one-off, development time
optimizations, have significant drawbacks limiting adoption in production
environments

Reply:
@svrnm what about that proposal 👆 ?

Reply:
Is there something we can group them by like

```
Traditional profiling solutions rely on instrumentation code that needs to be inserted during compilation. They have significant drawbacks ...
```

Reply:
"inserted during compilation" isn't true for all non-continuous profilers (e.g. profilers for scripted languages can be switched on/off via configuration or command-line flags). Also, you could have a continuous profiler that requires instrumentation.

Maybe we can summarize "traditional" as "(normally) require restarting of targeted apps or interpreters" or as "require changes of targeted apps" or "will interrupt production flow". You get the point, you'll surely find a better wording.